### PR TITLE
Added search for ZIP and supported candidate

### LINKED ISSUE #31: Dashboard: add search parameter by zipcode
Better if it's a range of zipcodes, for example I currently need to find someone who lives in LA

### LINKED ISSUE #35: Dashboard: add filter for who the person supported
e.g., I often just want to see the list of unpaired Trump supporters or unpaired Jill Stein supporters and match them

### PR DESCRIPTION
**Pull request by duncanka**
Resolves #35, and partially resolves #31 (searches for ZIP codes sharing the same first three digits; see comments on #31).

### CODE REVIEW

File: app/controllers/admin_controller.rb [hunk 1]
```diff
@@ -15,7 +15,22 @@ def index
...
       @title = 'Unpaired Users'
     end
-    scope = scope.fuzzy_search(params[:q]) if params[:q].present?
+
+    name_email = params[:name_email]
+    zip = params[:zip]
+    supported = params[:supported]
+
+    if name_email.present?
+         scope = scope.fuzzy_search(name_email) # Searches all string fields; presumably won't match ZIP
```
Comment (yasyf) [inline app/controllers/admin_controller.rb]: Stick these on one line if you can

`scope = scope.fuzzy_search(name_email) if name_email.present?`

File: app/controllers/admin_controller.rb [hunk 2]
```diff
@@ -15,7 +15,22 @@ def index
...
+    end
+    if zip.present?
+      # The first 3 digits of a ZIP code indicate a postal delivery region.
+      scope = scope.zip_starts_with(zip[0..2])
```
Comment (yasyf) [inline app/controllers/admin_controller.rb]: Same as above

File: app/controllers/admin_controller.rb [hunk 3]
```diff
@@ -15,7 +15,22 @@ def index
...
+    end
+    if supported.present?
+      scope = scope.where(supported: supported.to_i)
```
Comment (yasyf) [inline app/controllers/admin_controller.rb]: ditto

**Comment by yasyf** [comment]
👍  if you wouldn't mind fixing the one quick style thing

**Review by yasyf — CHANGES_REQUESTED** [review]
👍  if you wouldn't mind fixing the one quick style thing

**Review by yasyf — APPROVED** [review]

**Comment by yasyf** [comment]
deploying now

File: app/controllers/admin_controller.rb (hunk 1 above)
Comment (duncanka) [reply app/controllers/admin_controller.rb]: Oh, sure thing. I'm not usually a Ruby guy, so I'm sure I'll make plenty of style faux pas as I get up to speed.

**Pull request merged**